src/libstd/keyword_docs.rs

```diff
@@ -1167,12 +1182,53 @@ mod await_keyword {}
 
 #[doc(keyword = "dyn")]
 //
-/// Name the type of a [trait object].
+/// `dyn` is a prefix of a [trait object]'s type.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+/// Traits can be used as trait objects (via dynamic dispatch at runtime)
+/// or they can be used via generics or `impl Trait` parameters
+/// (monomorphization at compile time).
+///
+/// The `dyn` keyword is used to distinguish that a trait is being called by dynamic dispatch.
+/// As there is an additional runtime cost of calling a function on a trait object, the `dyn`
+/// keyword is used to highlight this cost.
+///
+/// For example, the following would pass in the [`Display`] trait as a trait object.
```

Comment:
A trait itself isn't what is being passed, but rather a value and a dictionary (well... a pointer to a dictionary... i.e. a pointer to a vtable).

Reply:
I also feel it would be clearer to invent a new trait here as opposed to using `Display` which requires you to talk about `println!` and that deeply "magical" infrastructure.

Reply:
I've dodged doing examples of each type of dispatch. There's better places for that I think.